test/test_helpers.py

```diff
@@ -0,0 +1,118 @@
+import dataclasses
+from collections import UserList
+from typing import Any
+from typing import Mapping
+from typing import Optional
+from typing import Sequence
+from typing import Union
+
+import more_itertools
+import pandas as pd
+import numpy as np
+from covidactnow.datapublic.common_fields import CommonFields
+from covidactnow.datapublic.common_fields import FieldName
+from covidactnow.datapublic.common_fields import PdFields
+
+from libs.datasets import timeseries
+from libs.pipeline import Region
+
+
+# This is a totally bogus fips/region/location that we've been using as a default in some test
+# cases. It is factored out here in an attempt to reduce how much it is hard-coded into our source.
+DEFAULT_FIPS = "97222"
+DEFAULT_REGION = Region.from_fips(DEFAULT_FIPS)
+
+
+class TimeseriesLiteral(UserList):
+    """Represents a timeseries literal, a sequence of floats and provenance string."""
+
+    def __init__(
+        self, ts_list, *, provenance: str = "",
+    ):
+        super().__init__(ts_list)
+        self.provenance = provenance
+
+
+def build_dataset(
+    metrics: Mapping[Region, Mapping[FieldName, Union[Sequence[float], TimeseriesLiteral]]],
```

Comment:
I think, `metrics_by_region` might be more helpful as to what the mapping is

Reply:
I thought the type make it fairly clear. Changed to `metrics_by_region_then_field_name`